Makefile.toml

```diff
@@ -48,6 +48,12 @@ dependencies = [
 command = "cargo"
 args = ["test", "--all-features", "--workspace", "--", "--nocapture"]
 
+[tasks.test-end-to-end]
+workspace = false
+script = '''
+./scripts/test-end-to-end.sh
```

Comment:
I usually think of E2E as a category of tests, and not a single test. I would call this more of a smoketest. But eventually we will need something more robust to run additional tests.

Reply:
The idea is that that the bash scripts can be extended with other flows enabling more testing opportunities. For now there is only 1 indeed.

Reply:
What I was trying to say with "something more robust" is that shell script is not a good language to write E2E tests. It works for a single test like this, which does a good job of covering the happy path and being a smoke test. But once we have a handful of tests we should be able to do other things, for example:

- have test selection, similar to how we do `cargo test -- <test_name>`
- have tests run in parallel, similar to how `cargo test` uses runners
- have the ability to debug the test scripts, and here shell scripts really lose the upper hand

To get the above, we will eventually have to translate the `.sh` into a `.rs` or `.py`. But I don't think we have to do this now.